pkgs/tools/security/yubihsm-shell/default.nix

```diff
@@ -0,0 +1,48 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, openssl
+, libusb1
+, libedit
+, curl
+, gengetopt
+, pkg-config
+, pcsclite
+, help2man
+}:
+
+stdenv.mkDerivation rec {
+  pname = "yubihsm-shell";
+  version = "2.3.1";
+
+  src = fetchFromGitHub {
+    owner = "Yubico";
+    repo = "yubihsm-shell";
+    rev = version;
+    sha256 = "sha256-D0kXiwc+i6mKA4oHuHjgXUmLMsmY5o/VI+1aCWtNjC0=";
+  };
+
+  nativeBuildInputs = [ pkg-config cmake help2man ];
+
+  buildInputs = [ libusb1 libedit curl gengetopt pcsclite openssl ];
+
+  # Can't find libyubihsm at runtime because of dlopen() in C code
+  postPatch = ''
+    substituteInPlace lib/yubihsm.c \
+      --replace "libyubihsm_usb.so" "$out/lib/libyubihsm_usb.so" \
+      --replace "libyubihsm_http.so" "$out/lib/libyubihsm_usb.so"
```

Comment:
Should not this be libyubihsm_http?

Reply:
Super derp if so. I will try with this fixed.

Reply:
Yup. That was it. How embarassing.. Let's pretend that never happened.